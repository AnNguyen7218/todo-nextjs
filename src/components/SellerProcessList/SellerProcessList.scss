@import 'config.gutters';
@import 'config.colors';
@import 'config.media';

.container {
  background-color: $light-gray;
  padding-top: gutter(48);
  padding-bottom: gutter(48);

  @include break(xl) {
    padding-top: gutter(96);
    padding-bottom: gutter(96);
  }
}

.desktop-container {
  display: flex;

  & > .sell-process-container {
    flex: 1;
    &:not(:first-child) {
      margin-left: gutter(24);

      @include break(xl) {
        margin-left: gutter(48);
      }
    }
  }
}

.header-container {
  margin-bottom: gutter(32);

  @include break(xl) {
    margin-bottom: gutter(48);
  }
}
