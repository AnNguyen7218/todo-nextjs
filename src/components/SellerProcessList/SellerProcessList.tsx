import React from 'react';
import { SELLER_PROCESSES } from 'constants/sellerLandingPage';
import { Container, SliderOnMobile, SellerProcess } from 'components';
import css from './SellerProcessList.scss';

const SellerProcessList = () => {
  const renderProcess = () => {
    return SELLER_PROCESSES.map((process, index) => {
      return (
        <div key={index} className={css['sell-process-container']}>
          <SellerProcess process={process} />
        </div>
      );
    });
  };

  const renderProcessDesktop = () => {
    return <div className={css['desktop-container']}>{renderProcess()}</div>;
  };

  return (
    <div className={css['container']}>
      <Container className={css['header-container']}>
        <h2>Get started in just {SELLER_PROCESSES.length} steps</h2>
      </Container>
      <SliderOnMobile
        desktopComponent={renderProcessDesktop()}
        mobileComponent={renderProcess()}
      />
    </div>
  );
};

export default SellerProcessList;
