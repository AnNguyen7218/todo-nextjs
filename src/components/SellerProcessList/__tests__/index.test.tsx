import React from 'react';
import { render, cleanup } from '@testing-library/react';
import SellerProcessList from '../SellerProcessList';

afterEach(cleanup);

describe('SellerProcessList Component', () => {
  test('SellerProcessList should match snapshot', () => {
    const { container } = render(<SellerProcessList />);
    expect(container).toMatchSnapshot();
  });
});
