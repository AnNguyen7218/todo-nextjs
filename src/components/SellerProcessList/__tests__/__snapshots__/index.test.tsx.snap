// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellerProcessList Component SellerProcessList should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="container header-container"
    >
      <h2>
        Get started in just 
        3
         steps
      </h2>
    </div>
    <div
      class="container desktop-container"
    >
      <div
        class="desktop-container"
      >
        <div
          class="sell-process-container"
        >
          <div
            class="container"
          >
            <img
              alt="process"
              class="img"
              src="step1.png"
            />
            <h3
              class="title"
            >
              01. Set up your account
            </h3>
            <p
              class="description"
            >
              Set up your account with key business information.
            </p>
          </div>
        </div>
        <div
          class="sell-process-container"
        >
          <div
            class="container"
          >
            <img
              alt="process"
              class="img"
              src="step2.png"
            />
            <h3
              class="title"
            >
              02. Complete your profile
            </h3>
            <p
              class="description"
            >
              Answer a few simple questions about your services, expertise, and preferences.
            </p>
          </div>
        </div>
        <div
          class="sell-process-container"
        >
          <div
            class="container"
          >
            <img
              alt="process"
              class="img"
              src="step3.png"
            />
            <h3
              class="title"
            >
              03. Create a service
            </h3>
            <p
              class="description"
            >
              Offer your professional services and start selling to customers all over the world.
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="container mobile-container"
    >
      <div
        class="swiper-container swiper-container-initialized swiper-container-horizontal"
      >
        <div
          class="swiper-wrapper"
          style="webkit-transition-duration: 0ms; transition-duration: 0ms; webkit-transform: translate3d(NaNpx, 0px, 0px); transform: translate3d(NaNpx, 0px, 0px);"
        >
          <div
            class="swiper-slide sell-process-container"
          >
            <div
              class="container"
            >
              <img
                alt="process"
                class="img"
                src="step1.png"
              />
              <h3
                class="title"
              >
                01. Set up your account
              </h3>
              <p
                class="description"
              >
                Set up your account with key business information.
              </p>
            </div>
          </div>
          <div
            class="swiper-slide sell-process-container"
          >
            <div
              class="container"
            >
              <img
                alt="process"
                class="img"
                src="step2.png"
              />
              <h3
                class="title"
              >
                02. Complete your profile
              </h3>
              <p
                class="description"
              >
                Answer a few simple questions about your services, expertise, and preferences.
              </p>
            </div>
          </div>
          <div
            class="swiper-slide sell-process-container"
          >
            <div
              class="container"
            >
              <img
                alt="process"
                class="img"
                src="step3.png"
              />
              <h3
                class="title"
              >
                03. Create a service
              </h3>
              <p
                class="description"
              >
                Offer your professional services and start selling to customers all over the world.
              </p>
            </div>
          </div>
        </div>
        <div
          class="swiper-pagination swiper-pagination-progressbar progress-bar"
        >
          <span
            class="progressbar-fill"
            style="webkit-transform: translate3d(0,0,0) scaleX(NaN) scaleY(1); transform: translate3d(0,0,0) scaleX(NaN) scaleY(1); webkit-transition-duration: 300ms; transition-duration: 300ms;"
          />
        </div>
        <span
          aria-atomic="true"
          aria-live="assertive"
          class="swiper-notification"
        />
      </div>
      <p
        class="count-slides"
      >
        01
         / 
        00
      </p>
    </div>
  </div>
</div>
`;
